src/aerie_cli/aerie_client.py

```diff
@@ -1442,6 +1539,22 @@ def delete_scheduling_goal(self, goal_id):
 
         return resp['id']
 
+    def delete_scheduling_goals(self, goal_id_list):
```

Comment:
I'll let @rohan-r-kulkarni give his reasoning but here's two off-the-cuff reasons to use a separate query
1. Speed. Doing a single network call to bulk-delete will always be faster than doing `n` network calls for each item of the list.
2. It makes the entire delete happen in a single transaction. If something goes wrong during this method (ie the transaction failed), _no work_ is applied. If this was to make the `n` calls to the single delete method, then a network failure _or_ transaction failure would cause part of the work to be applied.

Reply:
Thanks, that makes sense (especially #2). Maybe an alternate question is: could we just have a single query to upload one or more scheduling goals (& coalesce around that pattern in the future for all queries)? My angle is just that it would be nice to not have to maintain two nearly identical queries.

Reply:
Yeah, I agree that making the single-item function call the bulk function ought to be the pattern here.

Reply:
^^ Agreed. I also implemented a bulk upload, which addresses your comment on "upload_scheduling_goal**s**" vs. "upload_scheduling_goal". The plural "s" refers to a bulk upload/delete operation.

Reply:
Noted. I'll do this and add documentation for `upload_scheduling_goals` and `add_goals_to_specification` where the parameter was ambiguous.